tests/e2e/support/objects/app-files/share/actions.ts

```diff
@@ -60,21 +66,45 @@ export const createShare = async (args: createShareArgs): Promise<void> => {
 
 /**/
 
-export type ShareStatusArgs = Omit<ShareArgs, 'recipients'>
+export interface ShareStatusArgs extends Omit<ShareArgs, 'recipients'> {
+  via?: 'STATUS' | 'CONTEXT_MENU'
+}
 
 export const acceptShare = async (args: ShareStatusArgs): Promise<void> => {
-  const { resource, page } = args
-  await Promise.all([
-    page.locator(util.format(shareAcceptDeclineButton, resource, 'status-accept')).click(),
-    page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200),
-    page.locator(util.format(filesSharedWithMeAccepted, resource)).waitFor()
-  ])
+  const { resource, via, page } = args
+  if (via === 'CONTEXT_MENU') {
+    await clickActionInContextMenu({ page, resource }, 'accept-share')
+  } else {
+    await Promise.all([
+      page.locator(util.format(shareAcceptDeclineButton, resource, 'status-accept')).click(),
+      page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200)
+    ])
+  }
+  await page.locator(util.format(filesSharedWithMeAccepted, resource)).waitFor()
 }
 
 export const declineShare = async (args: ShareStatusArgs): Promise<void> => {
+  const { page, resource, via } = args
+  if (via === 'CONTEXT_MENU') {
+    await clickActionInContextMenu({ page, resource }, 'decline-share')
+  } else {
+    await page.locator(util.format(shareAcceptDeclineButton, resource, 'decline')).click()
+    await page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200)
+  }
+  await page.locator(util.format(filesSharedWithMeDeclined, resource)).waitFor()
+}
+
+export const clickActionInContextMenu = async (
+  args: ShareStatusArgs,
+  action: string
+): Promise<void> => {
   const { page, resource } = args
-  await page.locator(util.format(shareAcceptDeclineButton, resource, 'decline')).click()
-  await page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200)
+  await Promise.all([
+    page.locator(util.format(actionMenuDropdownButton, resource)).click(),
+    page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200),
+    page.locator(util.format(actionsTriggerButton, resource, action)).click(),
+    page.waitForResponse((resp) => resp.url().includes('shares') && resp.status() === 200)
```

Comment:
We have a lot of requests with `includes('shares') && resp.status() === 200` I would specify which request we check `DELETE` or `POST` (decline or accept)